### PR TITLE
Closes #254 Documentation of the Release to Github

### LINKED ISSUE #254: Documentation:  Clarification of Release Process on Github
### Please select a category the issue is focused on?

Other

### Let us know where something needs a refresh or put your idea here!

Add description of this process in [release strategy](https://pharmaverse.github.io/admiraldev/devel/articles/release_strategy.html)
Create the appropriate release tag and header for the release
image

Copy the News from the release
image

In the `Attach binaries by dropping them here or selecting them` upload the appropriate "admiral_x.x.x.tar.gz" file

### PR DESCRIPTION
**Pull request by zdz2101**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/devel/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiraldev/devel/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by zdz2101** [comment]
For easier review, preview of that snippet in screenshot below:

image

**Review by bms63 — COMMENTED** [review]
Could we also remove the `pre-release`  stuff?  I think we decided to remove this

File: vignettes/release_strategy.Rmd [hunk 1]
```diff
@@ -43,17 +43,17 @@ updates via the community meetings, especially any anticipated breaking changes.
...
+1) Use the release button on GitHub to "release" the package onto GitHub.  This release onto Github archives the version of code within the `main` branch, attaches the News/Changelog file, bundles the code into a `tar.gz` file and makes a validation report via the GitHub action `validation` from [insightsengineering/validatoR](https://github.com/insightsengineering/thevalidatoR).  Please see past [admiral releases](https://github.com/pharmaverse/admiral/releases) for reference and the [Releasing to Github](release_strategy.html#releasing-to-github) section for more details.
+1) Any issues fixed in the `main` branches should be merged back into `devel`. 
 1) Once a release is completed the `main` branch should be merged into `patch` to be ready for hotfixes.
 
 **Quarterly Release:**  
 
-* `devel >> pre-release >> main` 
-* `pre-release >> main >> devel` (Fixes done after a CRAN rejection)
+* `devel >> main` 
 * `main >> patch` (To be prepared in case of a needed hotfix)
```
Comment (bms63) [inline vignettes/release_strategy.Rmd:57]: I think an automated check PR will be developed whenever devel goes into main.  Can this be noted as a future upcoming feature to be on the look out for

File: vignettes/release_strategy.Rmd [hunk 2]
```diff
@@ -43,17 +43,17 @@ updates via the community meetings, especially any anticipated breaking changes.
...
-1) Once the package is available on CRAN, another Pull Request is created for merging the `pre-release` branch into the `main` branch.  This will trigger the GitHub action to rebuild the `{admiral}` website with all the updates for this release.
-1) Use the release button on GitHub to "release" the package onto GitHub.  This release onto Github archives the version of code within the `main` branch, attaches the News/Changelog file, bundles the code into a `tar.gz` file and makes a validation report via the GitHub action `validation` from [insightsengineering/validatoR](https://github.com/insightsengineering/thevalidatoR).  Please see past [admiral releases](https://github.com/pharmaverse/admiral/releases) for reference. 
-1) Any issues fixed in the `pre-release/main` branches should be merged back into `devel`. 
+1) Create a Pull Request from `devel` into the `main` branch.  Issues identified in this Pull Request should have work done in separate branches and merged once again into `devel`.
+1) Verify that all CI/CD checks are passing for the `devel` into the `main` Pull Request, merge and then bundle up and send off to CRAN.
```
Comment (bms63) [inline vignettes/release_strategy.Rmd:47]: https://r-pkgs.org/release.html#decide-the-release-type some of this is handy - can we point to this for additional reference

File: vignettes/release_strategy.Rmd [hunk 3]
```diff
@@ -43,17 +43,17 @@ updates via the community meetings, especially any anticipated breaking changes.
...
+1) If CRAN asks for modifications, repeat steps 1-2 as necessary.
+1) Once the package is accepted and available on CRAN, a GitHub action is set up to rebuild the `{admiral}` website with all the updates for this release.
+1) Use the release button on GitHub to "release" the package onto GitHub.  This release onto Github archives the version of code within the `main` branch, attaches the News/Changelog file, bundles the code into a `tar.gz` file and makes a validation report via the GitHub action `validation` from [insightsengineering/validatoR](https://github.com/insightsengineering/thevalidatoR).  Please see past [admiral releases](https://github.com/pharmaverse/admiral/releases) for reference and the [Releasing to Github](release_strategy.html#releasing-to-github) section for more details.
```
Comment (bms63) [inline vignettes/release_strategy.Rmd:50]: CAn we also point out that this action triggers a workflow to build the website

File: vignettes/release_strategy.Rmd [hunk 4]
```diff
@@ -63,12 +63,23 @@ Occasionally we will need to release a hot fix to address a package breaking bug
...
+1) Create a Pull Request from `patch` into the `main` branch.  Verify that all CI/CD checks are passing, merge and bundle up and send off to CRAN.
+1) If CRAN asks for modifications, repeat steps 1-4 as necessary.
+1) Use the release button on GitHub to "release" the package onto GitHub.  This release onto Github archives the version of code within the `main` branch, attaches the News/Changelog file, bundles the code into a `tar.gz` file and makes a validation report via the GitHub action `validation` from [insightsengineering/validatoR](https://github.com/insightsengineering/thevalidatoR).  Please see past [admiral releases](https://github.com/pharmaverse/admiral/releases) for reference and the [Releasing to Github](release_strategy.html#releasing-to-github) section for more details.
 1) These hot fixes should then be merged into the `devel` branch through an additional Pull Request.
```
Comment (bms63) [inline vignettes/release_strategy.Rmd:69]: Again release triggers website - could we have a screenshot showing 0.10.0 and the hotfix for 0.10.1???

**Comment by bms63** [comment]
Hey @zdz2101 did we finish this one?

**Pull request merged**